### PR TITLE
Reland #78164

### PR DESCRIPTION
**Pull request by robieta**
Reland https://github.com/pytorch/pytorch/pull/78164

Replaced `TEST_CUDA` with `torch.has_cuda`. The test in question was not expected to work with cuda; I was under the impression that `TEST_CUDA` would catch all the cases but it seems not.

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
LGTM if CI is green. 
P.S. You don't really need periodic for CPU tests, just trunk, don't you?

**Comment by robieta** [comment]
> LGTM if CI is green. 
It looks like the failures are infra rather than this PR, but I'll make sure it's 100% green before landing.

> P.S. You don't really need periodic for CPU tests, just trunk, don't you?
I couldn't find documentation for which label did what and the old `ci/all` listed these two as replacements. (And I figured better safe than sorry.)

**Comment by robieta** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/79175, since Ed wanted some changes and I think it's best to ghstack and land them together.

**Pull request closed without merging**
